dump-vtable.c

```diff
@@ -73,17 +73,38 @@ int get_VTables(char* file, vtable_info*** vtables)
         printf("WARNING Elf Library is out of date!\n");
     }
     
-    elf = elf_begin(fd, ELF_C_READ, NULL);
-    
+    if ((elf = elf_begin(fd, ELF_C_READ, NULL)) == NULL)
+    {
+        return ERR;
+    }
+
     while ((scn = elf_nextscn(elf, scn)) != NULL)
     {
-        gelf_getshdr(scn, &shdr);
+        if (gelf_getshdr(scn, &shdr) != &shdr)
+        {
+            return ERR;
+        }
         
         if (shdr.sh_type == SHT_DYNSYM)
         {
             edata = elf_getdata(scn, edata);
+            if (edata == NULL)
+            {
+                return ERR;
+            }
             
+            if (shdr.sh_entsize == 0)
+            {
+                return ERR;
+            }
+
+            // This could truncate on a 32bit machine inspecting a 64bit binary.
             symbol_count = shdr.sh_size / shdr.sh_entsize;
+
+            if (symbol_count > SIZE_MAX / sizeof(vtable_info*))
+            {
+                return ERR;
+            }
             *vtables = (vtable_info**)malloc(sizeof(vtable_info*) * symbol_count);
```

Comment:
Before, symbol_count was easy to corrupt. I think the guards help in this case. Double check these specifically.